tools/mpy-cross-all.py

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python3
+import argparse
+import os
+import os.path
+
+argparser = argparse.ArgumentParser(description="Compile all .py files to .mpy recursively")
+argparser.add_argument("-o", "--out", help="output directory (default: input dir)")
+argparser.add_argument("--target", help="select MicroPython target config")
+argparser.add_argument("-mcache-lookup-bc", action="store_true", help="cache map lookups in the bytecode")
+argparser.add_argument("dir", help="input directory")
+args = argparser.parse_args()
+
+TARGET_OPTS = {
+    "unix": "-mcache-lookup-bc",
+    "baremetal": "",
+}
+
+if not args.out:
+    args.out = args.dir
+
+path_prefix_len = len(args.dir) + 1
+
+for path, subdirs, files in os.walk(args.dir):
+    for f in files:
+        if f.endswith(".py"):
+            fpath = path + "/" + f
+            #print(fpath)
+            out_fpath = args.out + "/" + fpath[path_prefix_len:-3] + ".mpy"
+            out_dir = os.path.dirname(out_fpath)
+            if not os.path.isdir(out_dir):
+                os.makedirs(out_dir)
+            cmd = "mpy-cross -v -v %s %s -o %s" % (TARGET_OPTS.get(args.target, ""), fpath, out_fpath)
```

Comment:
Yep, I noticed too long paths, didn't give enough thought of how much to strip. Given that it repeats multiple times, even having an option to strip a barename might make sense. Will do as you suggest for now.